lms/djangoapps/course_api/views.py

```diff
@@ -2,52 +2,122 @@
 Course API Views
 """
 
-from rest_framework.views import APIView
-from rest_framework.response import Response
-from rest_framework.reverse import reverse
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.views import APIView, Response
 
+from opaque_keys import InvalidKeyError
 from opaque_keys.edx.keys import CourseKey
+
+
 from openedx.core.lib.api.view_utils import view_auth_classes
-from xmodule.modulestore.django import modulestore
+
+from .api import course_detail, list_courses
 
 
 @view_auth_classes()
-class CourseView(APIView):
+class CourseDetailView(APIView):
     """
-    Course API view
+    View class to describe a course
     """
 
     def get(self, request, course_key_string):
         """
         Request information on a course specified by `course_key_string`.
-            Body consists of a `blocks_url` that can be used to fetch the
-            blocks for the requested course.
+
+        Body consists of the following fields:
+            * `blocks_url`: can be used to fetch the blocks
+            * `course_image`:
+            * `description`:
+            * `end`: Date the course ends
+            * `enrollment_end`: Date enrollment ends
+            * `enrollment_start`: Date enrollment begins
+            * `id`: Course key
+            * `name`: Name of the course
+            * `number`: Catalog number of the course
+            * `org`: Name of the organization that owns the course
+            * `start`: Date the course begins
+            * `start_display`: Readably formatted start of the course
+            * `start_type`: Hint describing how `start_display` is set. One of:
+                  - `"string"`: manually set
+                  - `"timestamp"`: generated form `start` timestamp
+                  - `"empty"`: the start date should not be shown
 
         Arguments:
-            request (HttpRequest)
-            course_key_string
+            request: (HttpRequest)
+            course_key_string: (string) Key for the desired course
+
+        Parameters:
+            username (optional):
+                The username of the specified user whose visible courses we
+                want to see.  Defaults to the current user.
 
         Returns:
             HttpResponse: 200 on success
 
-
         Example Usage:
-
             GET /api/courses/v1/[course_key_string]
             200 OK
 
         Example response:
+            {
+                "blocks_url": "/api/courses/v1/blocks/?course_id=edX%2Ftoy%2F2012_Fall",
```

Comment:
@aleffert I'm not sure.  I haven't gone into pagination on this yet, so I'm not sure what other apis are doing.  One standard(ish) way to handle it would be to provide explicit `next` and `previous` links.  You'd know you were at the end because you didn't have a `next` link.  We could also provide a pagination section with total_count, page_number and related info.  But I'm just spitballing options.  I haven't actually looked to see if this is addressed anywhere.